synapse/config/_base.py

```diff
@@ -262,18 +262,23 @@ def read_templates(
         """
         search_directories = [self.default_template_dir]
 
-        # The loader will first look in the custom template directory (if specified) for the
-        # given filename. If it doesn't find it, it will use the default template dir instead
-        if custom_template_directory:
-            # Check that the given template directory exists
-            if not self.path_exists(custom_template_directory):
-                raise ConfigError(
-                    "Configured template directory does not exist: %s"
-                    % (custom_template_directory,)
-                )
+        # The loader will first look in the custom template directories (if specified)
+        # for the given filename. If it doesn't find it, it will use the default
+        # template dir instead.
+        if custom_template_directories is not None:
+            for custom_template_directory in custom_template_directories:
+                # Elements in the list might be None if they were retrieved from the
+                # configuration dict using config_dict.get(...).
+                if custom_template_directory:
+                    # Check that the given template directory exists
+                    if not self.path_exists(custom_template_directory):
+                        raise ConfigError(
+                            "Configured template directory does not exist: %s"
+                            % (custom_template_directory,)
+                        )
 
-            # Search the custom template directory as well
-            search_directories.insert(0, custom_template_directory)
+                    # Search the custom template directory as well
+                    search_directories.insert(0, custom_template_directory)
```

Comment:
this means we'll search the directories in reverse order to that given in the arguments, I think? Is that deliberate? It should probably be documented if so? If not, it suggests we need more tests?

Reply:
It was not deliberate, thanks for pointing it out! I've added a test that checks the order we search directories in.